gulpfile.js

```diff
@@ -467,6 +468,7 @@ function compileCss(watch) {
         return buildExtensions({
           bundleOnlyIfListedInFiles: false,
           compileOnlyCss: true,
+          compileAll: opt_compileAll,
```

Comment:
Adding the new option `compileAll ` is unnecessary. A simpler fix is to ignore `--extensions` and `--noextensions` if `compileOnlyCss` is `true`, since it's an indicator that we're calling `compileCss`. You could just change lines 261-264 to the following, and revert all the other changes.

```
    if (!options.compileOnlyCss &&
        extensionsToBuild.length > 0 &&
        extensionsToBuild.indexOf(extensions[key].name) == -1) {
      continue;
    }
```

Reply:
I used this approach first. But then I realized I am building css for all extensions even they are not required. I checked with `gulp` or `gulp build`, it's OK to skip building css and can skipping them help to a few seconds : )
Let me know if there's any reason that we should build all css no matter what. Thanks.

Reply:
Hmm. I think `gulp dist` needs the CSS in all extensions to be build, unlike `gulp build`. When I ran...
```
gulp clean
gulp dist --fortesting --noextensions
```
.. I saw these warnings that caused the build to fail:
```
Error compiling
extensions/amp-access/0.1/amp-access.js:23: WARNING - Failed to load module "../../../build/amp-access-0.1.css.js"
extensions/amp-ad/0.1/amp-ad.js:18: WARNING - Failed to load module "../../../build/amp-ad-0.1.css.js"
extensions/amp-form/0.1/amp-form.js:19: WARNING - Failed to load module "../../../build/amp-form-0.1.css.js"
extensions/amp-story/0.1/amp-story.js:35: WARNING - Failed to load module "../../../build/amp-story-0.1.css.js"
extensions/amp-user-notification/0.1/amp-user-notification.js:17: WARNING - Failed to load module "../../../build/amp-user-notification-0.1.css.js"
0 error(s), 5 warning(s)
```
Unless you can come up with a smaller list of extensions for which to build just the CSS? (I figured it takes less than 1 sec to build all the CSS, so why not just do it?)

Thoughts?

Reply:
`gulp dist` DO need all CSS to be built. but `gulp build` and `gulp watch` don't

I think the two approaches are really about whether we prefer to build all the CSS in all cases or we prefer an extra option `compileAll`. 

With an extra option `compileAll`, we only build all CSS with `gulp dist`, but NOT `gulp build` or `gulp watch`
With your proposed solution, we build all CSS with `gulp dist` `gulp build` AND `gulp watch`

TBH I feel either approach is fine : ) Let me know if you want me to switch.

Reply:
Okay, I'm convinced :) Thanks for patiently explaining. I retract all my other comments, but will leave them up for posterity.